lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -0,0 +1,213 @@
+            if (errorQueryParam === "signin") {
+                error = "Something went wrong. Please try again";
+            } else if (errorQueryParam === "restart") {
+                error = "The link you tried to click expired or was revoked. Please try again.";
+            }
+        }
+        this.state = {
+            loaded: false,
+            loginAttemptInfo: undefined,
+            error,
+        };
+    }
+
+    getIsEmbedded = (): boolean => {
+        if (this.props.isEmbedded !== undefined) {
+            return this.props.isEmbedded;
+        }
+        return false;
+    };
+
+    getModifiedRecipeImplementation = (): RecipeInterface => {
+        return {
+            ...this.props.recipe.recipeImpl,
+            createCode: async (input) => {
+                let contactInfo;
+                if ("email" in input) {
+                    contactInfo = input.email;
+                } else {
+                    contactInfo = formatPhoneNumberIntl(input.phoneNumber);
+                }
+                const res = await this.props.recipe.recipeImpl.createCode(input);
+                if (res.status === "OK") {
+                    const loginAttemptInfo = {
+                        ...res,
+                        lastResend: new Date().getTime(),
+                        contactInfoType: this.props.recipe.config.contactInfoType,
+                        contactInfo,
+                    };
+                    this.props.recipe.recipeImpl.setLoginAttemptInfo(loginAttemptInfo);
+                    this.setState((os) => ({
+                        ...os,
+                        error: undefined,
+                        loginAttemptInfo,
+                    }));
+                }
+                return res;
+            },
+            resendCode: async (input) => {
+                if (!this.state.loginAttemptInfo) {
+                    // This should never happen, but it makes TS happy
+                    throw new Error("Resend without loginAttemptInfo");
+                }
+                const res = await this.props.recipe.recipeImpl.resendCode(input);
+                if (res.status === "OK") {
+                    const loginAttemptInfo = {
+                        ...this.state.loginAttemptInfo,
+                        lastResend: new Date().getTime(),
+                    };
+                    this.props.recipe.recipeImpl.setLoginAttemptInfo(loginAttemptInfo);
+                    this.setState((os) => ({
+                        ...os,
+                        error: undefined,
+                        loginAttemptInfo,
+                    }));
+                } else if (res.status === "RESTART_FLOW_ERROR") {
+                    await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+
+                    this.setState((os) => ({
+                        ...os,
+                        error: "Please try again.",
+                        loginAttemptInfo: undefined,
+                    }));
```

Comment:
Shouldn't we redirect the user to the login page here? With an appropriate message.

Reply:
By removing the loginAttemptInfo you get back to the original mobile or email form.

Reply:
Let's discuss this:
- It's not obvious that this is happening, cause the recipeImplementation.ts file isn't doing that
- What will be the error message displayed to the user in that case?